Sources/PackageGraph/PackageGraphRoot.swift

```diff
@@ -47,14 +47,39 @@ public struct PackageGraphRoot {
     }
 
     /// The top level dependencies.
-    public let dependencies: [PackageDependency]
+    private let _dependencies: [PackageDependency]
+
+    public var dependencies: [PackageDependency] {
+        guard let mirrors = mirrors else {
+            return self._dependencies
+        }
+
+        return self._dependencies.map {
+            switch $0 {
+            case .fileSystem: return $0
+            case .registry: return $0
+            case .sourceControl(let settings):
+                let mappedLocation: PackageDependency.SourceControl.Location
+                switch settings.location {
+                case .local:
+                    mappedLocation = settings.location
+                case .remote(let url):
+                    let effectiveLocation = mirrors.effective(for: url.absoluteString)
```

Comment:
I think we only support mirrors for remote SCM dependencies. Is that right, @tomerd?

Reply:
yes, registry and remote scm

Reply:
OK, I need to handle the registry case here then.

Reply:
Actually looks like we support local SCM as well, this is the code we have in the JSON parser (!):

```
location = identityResolver.mappedLocation(for: location)
if PackageIdentity.plain(location).isRegistry {
	// re-mapped to registry
       [...]
} else if let localPath = try? AbsolutePath(validating: location) {
    // a package in a git location, may be a remote URL or on disk
    // in the future this will check with the registries for the identity of the URL
   [...]
} else {
    let url = SourceControlURL(location)
    [...]
}
```

It does seem a bit odd that the code paths for `parseSourceControlDependency()` and `parseRegistryDependency()` diverge. @tomerd is it intentional that registry dependencies only support mirroring to remote SCM but not local which is different from SCM dependencies?

I feel like ideally we could unify these two code paths into a single one and re-use for remapping the extra dependencies, too, but if there are intentional differences here that will be more difficult.

Reply:
> I feel like ideally we could unify these two code paths into a single one and re-use for remapping the extra dependencies, too, but if there are intentional differences here that will be more difficult.

cant remember the details, but +1 to try and consolidate and reduce divergence

Reply:
iirc the mirrors only allow remote scm -> remote scm, registry -> registry, remote scm -> registry, and registry -> remote scm. you cannot map to a local afair

Reply:
Right now, remote SCM also allows local SCM (and presumably vice versa), but registry does not allow local SCM. If we don't know of a strong reason for that, I'm just going to unify these to both allow local SCM. Local paths are not allowed at all, I think we should keep it that way.